arbstate/inbox.go

```diff
@@ -41,6 +44,29 @@ type MessageWithMetadata struct {
 	DelayedMessagesRead uint64                   `json:"delayedMessagesRead"`
 }
 
+var EmptyTestMessageWithMetadata = MessageWithMetadata{
+	Message: &arbos.EmptyTestIncomingMessage,
+}
+
+// Message signature is only verified if requestId defined
+var TestMessageWithMetadataAndRequestId = MessageWithMetadata{
+	Message: &arbos.TestIncomingMessageWithRequestId,
+}
+
+func (m *MessageWithMetadata) Hash(sequenceNumber arbutil.MessageIndex, chainId uint64) (common.Hash, error) {
+	serializedExtraData := make([]byte, 24)
+	binary.BigEndian.PutUint64(serializedExtraData[:8], uint64(sequenceNumber))
+	binary.BigEndian.PutUint64(serializedExtraData[8:16], chainId)
+	binary.BigEndian.PutUint64(serializedExtraData[16:], m.DelayedMessagesRead)
+
+	serializedMessage, err := m.Message.Serialize()
+	if err != nil {
+		return common.Hash{}, errors.Wrapf(err, "unable to serialize message %v", sequenceNumber)
```

Comment:
Does this handle messages without a request id?

Reply:
I think we should use rlp encoding here instead, to avoid writing a second serialization format.